imageio/plugins/ffmpeg.py

```diff
@@ -512,7 +514,8 @@ def _load_infos(self):
             if match is not None:
                 hms = map(float, line[match.start()+1:match.end()].split(':'))
                 self._meta['duration'] = duration = cvsecs(*hms)
-                self._meta['nframes'] = int(round(duration*fps))
+                if fps:
+                    self._meta['nframes'] = int(round(duration*fps))
```

Comment:
Perhaps this line is causing many of the off by one errors? Not to mention there are files for which fps is not constant (https://github.com/imageio/imageio/issues/176).